app/controllers/application_controller.rb

```diff
@@ -1,10 +1,15 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :create_cart
```

Comment:
This was my proposal but I'm open to other implementations. Go ahead and work with @Scharrre97 and @fabianlundgren 
As to the performance issues this method is only hitting the db once per session. All other times it only checks the session and that is not performance heavy.